consensus/broadcast/shardChainMessenger_test.go

```diff
@@ -375,6 +399,19 @@ func TestShardChainMessenger_BroadcastHeaderNilHeaderShouldErr(t *testing.T) {
 	assert.Equal(t, spos.ErrNilHeader, err)
 }
 
+func TestShardChainMessenger_BroadcastHeaderNilHeaderShouldErrMockMarshalizer(t *testing.T) {
+	marshalizer := mock.MarshalizerMock{
+		Fail: true,
+	}
+
+	args := createDefaultShardChainArgs()
+	args.Marshalizer = marshalizer
+	scm, _ := broadcast.NewShardChainMessenger(args)
+
+	err := scm.BroadcastHeader(&block.MetaBlock{Nonce: 10}, []byte("pk bytes"))
+	assert.Equal(t, mock.ErrMockMarshalizer, err)
```

Comment:
the test mentions that with nil header it should fail, but it doesn't test that, the passed header is not nil, and the asserted error is ErrMockMarshalizer instead of ErrNilHeader